workflows/prognostic_c48_run/runtime/segmented_run/logs.py

```diff
@@ -0,0 +1,144 @@
+
+
+@dataclass
+class LogLine:
+    type: LineType
+    data: dict
+    line: str
+
+
+def parse_line(line):
+    for line_type in LineType:
+        match = REGEX[line_type].match(line)
+        if match:
+            return LogLine(line_type, match.groupdict(), line)
+    raise ValueError(f"Unable to parse line: {line}")
+
+
+class Handler:
+    """Handler handels the parsed lines produces by the log stream
+
+    Attributes:
+        model_time: the last handled time
+    """
+
+    LABEL_NAME = "logging.googleapis.com/labels"
+
+    def __init__(self, labels=Mapping[str, Any]):
+        """
+        Args:
+            labels: labels to make available for query in the Google Cloud Logging
+                console
+        """
+        self.model_time: Optional[str] = None
+        self.labels = labels
+
+    def _insert_model_time(self, payload):
+        payload[self.LABEL_NAME]["model_time"] = self.model_time
+        payload["model_time"] = self.model_time
+
+    def handle(self, line: LogLine):
+        payload = {**line.data}
+        payload[self.LABEL_NAME] = {"kind": line.type.value, **self.labels}
+        if line.type == LineType.PYTHON_LOG:
+            message = payload.pop("message")
+            try:
+                payload["json"] = json.loads(message)
+            except json.JSONDecodeError:
+                payload["message"] = message
+
+        if line.type == LineType.FV3_LOG:
+            payload["severity"] = "DEBUG"
+
+        self.model_time = payload.get("json", {}).get("time") or self.model_time
+
+        self._insert_model_time(payload)
+        return json.dumps(payload)
+
+
+def handle_fv3_log(f: Iterable[str], labels: Mapping[str, Any] = {},) -> Iterable[str]:
+    """Consume a fv3 log stream and produce a json stream
+
+    Args:
+
+        f: a iterable of lines ... e.g. a file object or sys.stdin
+        labels: the labels to make available for query in the Google Cloud Logging
+        output_stream: where to write the transformed data to e.g. sys.stdout
+
+    Examples:
+
+        Here are some jq filters that can processe these outputs.
+
+        To print the fortran log lines only::
+
+            jq -r 'select(.["logging.googleapis.com/labels"].kind == "fortran" ) \
+                | .message'
+
+        To print just the water vapor path and time::
+
+            jq -r 'select(.json.water_vapor_path )  \
+            | {time: .model_time, wvp: .json.water_vapor_path}'
+
+        To show all min/max variables::
+
+            jq -sr '[.[].max_min_name | select(. != null) | . ]  |  unique'
+
+    """  # noqa
+    handler = Handler(labels)
+    for line in f:
+        yield handler.handle(parse_line(line))
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG)
+    fname = "/Users/noahb/workspace/ai2cm/fv3net/logs.txt"
+    import fileinput
+
+    with fileinput.input() as f:
+        for line in handle_fv3_log(f, labels={"fname": fileinput.filename()}):
+            print(line)
```

Comment:
looks like leftover testing code?

Reply:
It's a useful cli for now. It'd be nice to expose this as a public CLI...and extract it and the other runfv3 code from the prognostic run.

Reply:
I added this as a CLI `runfv3 parse-logs`.